### PR TITLE
new switches to override avatarURL

### PR DESCRIPTION
**Pull request by davidkelly**
We have 2 new command-line switches:  --avatarURL and --replaceAvatarURL.  The former uses the url specified as your avatarURL for this session only, the latter also persists it.  Note in either case, you are free to change it at any time, and that will persist.

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan:

launch interface, set your avatarURL to blank
stop interface
start it again, but with --avatarURL <some valid avatar url>
you (and others) should see you as that avatar 
stop interface
restart interface, with nothing on the command-line
you (and others) should see you as the default avatar
stop interface
start interface, with --replaceAvatarURL <some avatar url>
again, you (and others) should see you as this avatar
stop interface
start interface, with nothing on the command-line
you (and others) should see you as that avatar you set above, not the default avatar
:beers:

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -859,6 +859,17 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
     }
     ResourceCache::setRequestLimit(concurrentDownloads);
 
+    // perhaps override the avatar url.  We test later for validity, so
+    // no need to do so here.  If someone specifies both --avatarURL
+    // and --replaceAvatarURL, the replaceAvatarURL wins.
+    QString avatarURL = getCmdOption(argc, constArgv, "--avatarURL");
+    _avatarOverrideUrl = QUrl::fromUserInput(avatarURL);
```
Comment (zfox23) [inline interface/src/Application.cpp]: Any reason you didn't wrap this in `if (!replaceURL.isEmpty())` like below?

Comment (davidkelly) [inline interface/src/Application.cpp]: turns out I test for isValid before using, and isValid is false if empty.  One less if statement.

Comment (zfox23) [inline interface/src/Application.cpp]: Nice.

Comment (zfox23) [inline interface/src/Application.cpp]: Got it, yeah - I'd prefer if the comment were a bit more clear. But I understand now.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/974/HighFidelity-Beta-PR10776-92cbb86e8ba512de713a76b3432b5bea51d79d39.dmg

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -929,10 +929,14 @@ void MyAvatar::saveData() {
 
     settings.setValue("scale", _targetScale);
 
-    settings.setValue("fullAvatarURL",
+    // only save the fullAvatarURL if it has not been overwritten on command line, or it
+    // has but we want it saved
```
Comment (zfox23) [inline interface/src/avatar/MyAvatar.cpp]: This comment is also a little confusing to me. I now understand how the comment relates to the logic, but I won't in a couple of weeks 😄 perhaps:
`// Only save the fullAvatarURL if "--replaceAvatarURL" argument was specified or if "_avatarOverrideUrl" is invalid/empty`

?

Comment (davidkelly) [inline interface/src/avatar/MyAvatar.cpp]: I suppose that is true.  I'll update the comment

**Comment by CainFoool** [comment]
Testing.

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/984/HighFidelity-Beta-PR10776-92cbb86e8ba512de713a76b3432b5bea51d79d39.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/984/HighFidelity-Beta-PR10776-92cbb86e8ba512de713a76b3432b5bea51d79d39.dmg

**Comment by CainFoool** [comment]
# Test Passed 
Test | Result
-------- | --------
launch interface, set your avatarURL to blank | Pass
stop interface | Pass
start it again, but with --avatarURL | Pass
you (and others) should see you as that avatar | Pass
stop interface | Pass
restart interface, with nothing on the command-line | Pass
you (and others) should see you as the default avatar | Pass
stop interface | Pass
start interface, with --replaceAvatarURL | Pass
again, you (and others) should see you as this avatar | Pass
stop interface | Pass
start interface, with nothing on the command-line | Pass
you (and others) should see you as that avatar you set above, not the default avatar | Pass

Let me know if you'd like anything else tested.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by davidkelly** [comment]
test this please

**Comment by davidkelly** [comment]
updated comments, so this looks ready to merge to me

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/995/HighFidelity-Beta-PR10776-5da01f00743d34c030cbb9dfbd9926616e32eccb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/995/HighFidelity-Beta-PR10776-5da01f00743d34c030cbb9dfbd9926616e32eccb.exe

**Pull request merged**